sql/src/main/java/io/crate/protocols/postgres/types/AnyType.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import io.netty.buffer.ByteBuf;
+
+import javax.annotation.Nonnull;
+import java.nio.charset.StandardCharsets;
+
+public class AnyType extends PGType<Integer> {
+    // we represent the any type as integer,
+    // such as the typalign of any is integer
```

Comment:
Can you add a reference to where this is coming from? 

Would also rename `such as` to `Because typalign of any is integer`

Reply:
Could it be that the `int` that is streamed as part of any is the `oid` of the actual type, and that `any` is some kind of "lazy" / deferred type?

Reply:
I'll check that. 

I couldn't find anything so far in the documentation, I just inferred it from the `typalign` of the PostgreSQL `any` type.

Reply:
Yes, it seems that the type of these pseudotypes is defined by a data type that is passed to it later on. See, https://www.postgresql.org/docs/10/extend-type-system.html